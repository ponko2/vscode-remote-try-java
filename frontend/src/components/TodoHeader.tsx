import clsx from "clsx";
import type React from "react";

type Props = {
  todosCount: number;
  completedTodosCount: number;
  createTodo: (title: string) => void;
  toggleAllTodos: () => void;
};

export function TodoHeader({
  todosCount,
  completedTodosCount,
  createTodo,
  toggleAllTodos,
}: Props) {
  function handleKeyDown(event: React.KeyboardEvent<HTMLInputElement>) {
    const newValue = event.currentTarget.value.trim();
    if (event.key === "Enter") {
      if (newValue.length !== 0) {
        createTodo(newValue);
      }
      event.currentTarget.value = "";
    }
  }

  return (
    <header className={clsx("relative", "mt-32", "h-16")}>
      <h1
        className={clsx(
          "absolute",
          "bottom-16",
          "w-full",
          "pb-6",
          "text-center",
          "text-7xl/none",
          "font-extralight",
          "text-red-700",
          "[text-rendering:optimizeLegibility]",
        )}
      >
        todos
      </h1>
      <input
        className={clsx(
          "size-full",
          "py-4",
          "pl-14",
          "pr-4",
          "text-2xl",
          "shadow-inner",
          "placeholder:font-normal",
          "placeholder:italic",
          "placeholder:text-black/40",
          "focus:shadow",
          "focus:shadow-red-400",
          "focus:outline-none",
        )}
        onKeyDown={handleKeyDown}
        placeholder="What needs to be done?"
        type="text"
      />
      {!!todosCount && (
        <label>
          <input
            checked={completedTodosCount === todosCount}
            className={clsx("peer", "appearance-none")}
            onClick={toggleAllTodos}
            readOnly
            type="checkbox"
          />
          <span
            className={clsx(
              "absolute",
              "left-0",
              "top-0",
              "flex",
              "h-full",
              "w-12",
              "items-center",
              "justify-center",
              "text-[0]",
              "before:inline-block",
              "before:rotate-90",
              "before:px-7",
              "before:py-2.5",
              "before:text-2xl",
              "before:text-neutral-400",
              "before:content-['❯']",
              "peer-checked:before:text-neutral-700",
              "peer-focus:shadow",
              "peer-focus:shadow-red-400",
              "peer-focus:outline-none",
            )}
          >
            Mark all as complete
          </span>
        </label>
      )}
    </header>
  );
}
