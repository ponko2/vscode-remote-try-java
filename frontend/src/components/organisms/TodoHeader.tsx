import React from "react";
import styles from "./TodoHeader.module.css";

type Props = {
  todosCount: number;
  completedCount: number;
  addTodo: (title: string) => void;
  toggleAllTodo: () => void;
};

export const TodoHeader = ({
  todosCount,
  completedCount,
  addTodo,
  toggleAllTodo,
}: Props): JSX.Element => {
  function handleKeyDown(event: React.KeyboardEvent<HTMLInputElement>) {
    const newValue = event.currentTarget.value.trim();
    if (event.key === "Enter") {
      if (newValue.length !== 0) {
        addTodo(newValue);
      }
      event.currentTarget.value = "";
    }
  }

  return (
    <header className={styles.header}>
      <h1>todos</h1>
      <input
        type="text"
        className={styles.newTodo}
        placeholder="What needs to be done?"
        onKeyDown={handleKeyDown}
      />
      {!!todosCount && (
        <label className={styles.toggleAll}>
          <input
            type="checkbox"
            checked={completedCount === todosCount}
            onClick={toggleAllTodo}
            readOnly
          />
          <span>Mark all as complete</span>
        </label>
      )}
    </header>
  );
};
