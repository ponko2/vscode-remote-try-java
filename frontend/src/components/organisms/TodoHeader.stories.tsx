import { action } from "@storybook/addon-actions";
import type { Meta, StoryObj } from "@storybook/react";
import { TodoHeader } from "./TodoHeader";

const meta = {
  component: TodoHeader,
} satisfies Meta<typeof TodoHeader>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Basic = {
  args: {
    addTodo: action("addTodo"),
    toggleAllTodo: action("toggleAllTodo"),
    todosCount: 0,
    completedCount: 0,
  },
} satisfies Story;

export const HasActive = {
  args: {
    ...Basic.args,
    todosCount: 1,
    completedCount: 0,
  },
} satisfies Story;

export const IsAllCompleted = {
  args: {
    ...Basic.args,
    todosCount: 1,
    completedCount: 1,
  },
} satisfies Story;
