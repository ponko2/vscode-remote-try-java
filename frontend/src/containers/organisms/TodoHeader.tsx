import { TodoHeader as Component } from "../../components/organisms/TodoHeader";
import {
  useAddTodo,
  useCompletedCount,
  useTodosCount,
  useToggleAllTodo,
} from "../../hooks/todos";

export function TodoHeader(): JSX.Element {
  const { data: todosCount = 0 } = useTodosCount();
  const { data: completedCount = 0 } = useCompletedCount();
  const addTodo = useAddTodo();
  const toggleAllTodo = useToggleAllTodo();
  return (
    <Component
      todosCount={todosCount}
      completedCount={completedCount}
      addTodo={addTodo.mutate}
      toggleAllTodo={toggleAllTodo.mutate}
    ></Component>
  );
}
