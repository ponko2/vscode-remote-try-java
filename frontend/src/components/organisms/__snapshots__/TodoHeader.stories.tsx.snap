// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/organisms/TodoHeader Basic smoke-test 1`] = `
<header class="_header_vkyfp_1">
  <h1>
    todos
  </h1>
  <input type="text"
         class="_newTodo_vkyfp_16"
         placeholder="What needs to be done?"
  >
</header>
`;

exports[`components/organisms/TodoHeader HasActive smoke-test 1`] = `
<header class="_header_vkyfp_1">
  <h1>
    todos
  </h1>
  <input type="text"
         class="_newTodo_vkyfp_16"
         placeholder="What needs to be done?"
  >
  <label class="_toggleAll_vkyfp_40">
    <input type="checkbox"
           readonly
    >
    <span>
      Mark all as complete
    </span>
  </label>
</header>
`;

exports[`components/organisms/TodoHeader IsAllCompleted smoke-test 1`] = `
<header class="_header_vkyfp_1">
  <h1>
    todos
  </h1>
  <input type="text"
         class="_newTodo_vkyfp_16"
         placeholder="What needs to be done?"
  >
  <label class="_toggleAll_vkyfp_40">
    <input type="checkbox"
           readonly
           checked
    >
    <span>
      Mark all as complete
    </span>
  </label>
</header>
`;
